turbo.json

```diff
@@ -0,0 +1,30 @@
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "build:babel": {
+      "outputs": ["build/**/*", "build-cjs/**/*"]
+    },
+    "//#build:typecheck:tsc": {
```

Comment:
unlike other commands which are all run once per app/package, this one is run from the root. That explains the weird syntax and the fact we had to define the `inputs` as turbo wasn't able to figure it out so easily and this caused issues with caching. 

Would be nice to break this job down so each app and package has its own typecheck but for now that should do.